integration/simulation/transaction_injector.go

```diff
@@ -192,28 +233,91 @@ func (m *TransactionInjector) issueRandomTransfers() {
 }
 
 // issueRandomDeposits creates and issues a number of transactions proportional to the simulation time, such that they can be processed
-// Generates L1 common.DepositTx transactions
+// Generates L1 common.L1DepositTx transactions
 func (m *TransactionInjector) issueRandomDeposits() {
 	for ; atomic.LoadInt32(m.interruptRun) == 0; time.Sleep(obscurocommon.RndBtwTime(m.avgBlockDuration, m.avgBlockDuration*2)) {
 		v := obscurocommon.RndBtw(1, 100)
-		txData := obscurocommon.L1TxData{
-			TxType: obscurocommon.DepositTx,
-			Amount: v,
-			Dest:   rndWallet(m.wallets).Address,
+		addr := m.ethWallet.Address()
+		txData := &obscurocommon.L1DepositTx{
+			Amount:        v,
+			To:            m.mgmtContractAddr,
+			TokenContract: m.erc20ContractAddr,
+			Sender:        &addr,
 		}
-		m.rndL1Node().BroadcastTx(&txData)
+		tx := m.erc20ContractLib.CreateDepositTx(txData, m.ethWallet.GetNonceAndIncrement())
+		signedTx, err := m.ethWallet.SignTransaction(tx)
+		if err != nil {
+			panic(err)
+		}
+		err = m.rndL1Node().IssueTransaction(signedTx)
+		if err != nil {
+			panic(err)
+		}
+
 		m.stats.Deposit(v)
 		go m.trackL1Tx(txData)
 	}
 }
 
+// issueRandomERC20Deposits creates and issues a number of transactions proportional to the simulation time, such that they can be processed
+// Generates L1 common.L1DepositTx transactions
+func (m *TransactionInjector) issueRandomERC20Deposits() {
```

Comment:
I don't understand what the difference behind `issueRandomERC20Deposits` and `issueRandomDeposits`  is

Reply:
Good point, updated those, don't make sense anymore. Please review